packages/dds/ordered-collection/src/consensusOrderedCollection.ts

```diff
@@ -327,7 +327,9 @@ export class ConsensusOrderedCollection<T = any>
             }
             if (local) {
                 assert(
-                    localOpMetadata, `localOpMetadata is missing from the local client's ${op.opName} operation`);
+                    // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
```

Comment:
I'd change cast below to include undefined and move this assert after it, testing resolve for not being undefined.
That said, maybe even better - just remove assert, it does not add much value - we will crash one way or another if things go wrong